src/KubernetesClient.Models/KubernetesYaml.cs

```diff
@@ -15,16 +15,22 @@ namespace k8s
     /// </summary>
     public static class KubernetesYaml
     {
-        private static readonly IDeserializer Deserializer =
+        private static readonly DeserializerBuilder CommonDeserializerBuilder =
             new DeserializerBuilder()
                 .WithNamingConvention(CamelCaseNamingConvention.Instance)
                 .WithTypeConverter(new IntOrStringYamlConverter())
                 .WithTypeConverter(new ByteArrayStringYamlConverter())
                 .WithTypeConverter(new ResourceQuantityYamlConverter())
                 .WithAttemptingUnquotedStringTypeDeserialization()
-                .WithOverridesFromJsonPropertyAttributes()
-                .IgnoreUnmatchedProperties()
-                .Build();
+                .WithOverridesFromJsonPropertyAttributes();
+        private static readonly IDeserializer StrictDeserializer =
+            CommonDeserializerBuilder
+            .Build();
+        private static readonly IDeserializer Deserializer =
+            CommonDeserializerBuilder
+            .IgnoreUnmatchedProperties()
+            .Build();
+        private static IDeserializer GetDeserializer(bool strict) => strict ? StrictDeserializer : Deserializer;
```

Comment:
still thinking about the better way to inject something `staticly`

Reply:
The problem is the mix of two design patterns - factory pattern of the Yaml (de)serializer with stateless static API call from K8s client...